src/main/java/org/opentripplanner/middleware/auth/Auth0Connection.java

```diff
@@ -143,6 +143,25 @@ public static void checkUserIsAdmin(Request req, Response res) {
         }
     }
 
+    /**
+     * Check that an {@link ApiUser} is linked to an Api key.
+     */
+    //FIXME: Move this check into existing auth checks so it would be carried out automatically prior to any
+    // business logic. Consider edge cases where a user can be both an API user and OTP user.
+    public static void linkApiKeyToApiUser(Request req) {
```

Comment:
With respect to the method name, maybe just `ensureApiUserHasApiKey`? Has this logic changed from its previous iteration?

Javadoc: 
```
Check that the API key used in the incoming request is associated with the matching {@link ApiUser} (which is determined from the Authorization header).